draft-ietf-taps-transport-security.md

```diff
@@ -854,170 +354,18 @@ configure and works over a variety of transports. OpenVPN encapsulates either
-their presence impacts transport services or if a necessary transport service or application dependency
-is unavailable.
-
-In this context, an application dependency is an aspect of the security feature which can be exposed
-to the application. An application dependency may be required for the security feature to function,
-or it may provide additional information and control to the application. For example, an application
-may need to provide information such as keying material or authentication credentials, or it may want
-to restrict which cryptographic algorithms to allow for negotiation.
-
-## Mandatory Features
-
-Mandatory features must be supported regardless of transport and application services available. Note that not
-all mandatory features are provided by each surveyed protocol above. For example, tcpcrypt does not provide
-responder authentication and CurveCP does not provide forward-secure session key establishment.
-
-- Record or datagram confidentiality and integrity.
-  - Application dependency: None.
-
-- Forward-secure session key establishment.
-  - Application dependency: None.
-
-- Unilateral responder authentication.
-  - (Optional) Application dependency: Application-provided trust information. System trust stores may also be used to authenticate responders.
-
-## Optional Features
-
-In this section we list optional features along with their necessary application dependencies, if any.
-
-- Pre-shared key support (PSK):
-  - Application dependency: Application provisioning and distribution of pre-shared keys.
-
-- Mutual authentication (MA):
-  - Application dependency: Mutual authentication credentials required.
-
-- Cryptographic algorithm negotiation (AN):
-  - Application dependency: Application awareness of supported or desired algorithms.
-
-- Application authentication delegation (AD):
-  - Application dependency: Application opt-in and policy for endpoint authentication.
-
-- DoS mitigation (DM):
-  - Application dependency: None.
-
-- Connection mobility (CM):
-  - Application dependency: None.
-
-- Source validation (SV):
-  - Application dependency: None.
-
-- Application-layer feature negotiation (AFN):
-  - Application dependency: Specification of application-layer features or functionality.
-
-- Configuration extensions (CX):
-  - Application dependency: Specification of application-specific extensions.
-
-- Session caching and management (SC):
-  - Application dependency: None.
-
-- Length-hiding padding (LHP):
-  (Optional) Application dependency: Knowledge of desired padding policies. Some protocols, such as IKE, can negotiate application-agnostic padding policies.
-
-- Early data support (ED):
-  - Application dependency: Anti-replay protections or hints of data idempotency.
-
-- Record replay prevention (RP):
-  - Application dependency: None.
-
-- Out-of-order receipt record (OO):
-  - Application dependency: None.
-
-## Optional Feature Availability
+# Transport Dependencies
```

Comment:
Some of these protocols can run over either Stream or Datagram protocols, right? I suggest to add some nuance here in the future, and before we actually write that text, maybe keep the "Protocol dependencies" (see comment above).

Reply:
👍

Reply:
Filled out this section.